src/sentry/south_migrations/0455_create_global_searches.py

```diff
@@ -0,0 +1,1269 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+
+from sentry.models.savedsearch import DEFAULT_SAVED_SEARCHES
+
+
+class Migration(DataMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = False
+
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+        "Write your forwards methods here."
+        # Note: Don't use "from appname.models import ModelName".
+        # Use orm.ModelName to refer to models in this application,
+        # and orm['appname.ModelName'] for models in other applications.
+        SavedSearch = orm['sentry.SavedSearch']
+
+        for search in DEFAULT_SAVED_SEARCHES:
```

Comment:
i feel like we maybe discussed this in a meeting and i'm not remembering, but can you give a bit of context as to why we're saving these in the database? is it just what we did for project level searches?

Reply:
I don't think we ever discussed this in a meeting, it just seemed like the simplest model. It makes it simpler for the frontend to consume, since we can treat them all the same way.

When we click a saved search, we go to a url with the saved id. So if we save the global searches to the DB then it's easier to just keep that pattern the same. Otherwise we can return virtual rows (I did so in an earlier version of this), but then since they have no ids we can't use the same url pattern.

An alternative could be to return specific reserved ids for the virtual rows, or just have the frontend use a different url path and just pass the encoded query to it.

Reply:
ah yeah the saved id makes sense, thanks!